### DISCUSSION TITLE
[Documentation provided] How to install on Ubuntu 24.04?

### DISCUSSION THREAD

**Discussion by SamuelMarks** [containers/podman#22921] [General]
### Q: How do I install on Ubuntu 24.04

Install fails with `apt install podman`; and besides that an old version https://packages.ubuntu.com/noble/podman

### A: Solution

```sh
$ sudo apt install qemu-utils qemu-system-x86
$ curl -OL https://github.com/containers/gvisor-tap-vsock/releases/download/v0.7.3/gvproxy-linux-amd64
# ^ or a newer from https://github.com/containers/gvisor-tap-vsock/releases
$ sudo mkdir /usr/local/lib/podman
$ sudo install -m 0755 gvproxy-linux-amd64 /usr/local/lib/podman/gvproxy
$ curl -OL https://github.com/containers/podman/releases/download/v5.1.1/podman-remote-static-linux_amd64.tar.gz
# ^ or a newer from https://github.com/containers/podman/releases
$ tar xf podman-remote-static-linux_amd64.tar.gz
$ mv bin/* # choose a binary directory in your PATH as the destination
```

Then you should be able to run these two without error:
```sh
$ podman machine init
$ podman machine start
```

**Comment by afbjorklund** [comment]
The last command needs to rename `bin/podman-remote-static-linux_amd64` to `podman`, the same way as with `gvproxy`?

Or just keep the real name "podman-remote-static", if you want to use "podman" and "podman-remote" from the system.

* https://github.com/containers/podman-desktop/discussions/5762

* https://github.com/containers/podman/discussions/17362

**Comment by afbjorklund** [comment]
> Install fails with `apt install podman`

This should be reported to Ubuntu

**Reply by SamuelMarks** [reply]
```
$ sudo apt install podman
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package podman is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'podman' has no installation candidate
```

**Reply by afbjorklund** [reply]
Weird, it seems to resolve just fine from a `docker.io/library/ubuntu:24.04` container

```console
root@800f6dbb0b2d:/# apt-cache madison podman
    podman | 4.9.3+ds1-1build2 | http://archive.ubuntu.com/ubuntu noble/universe amd64 Packages
```

**Reply by SamuelMarks** [reply]
Confirmed it's not on a 20.04 box also

**Reply by afbjorklund** [reply]
It was not included in 20.04, only in 20.10

**Comment by jdkruzr** [comment]
anyone know how to get Podman 5 installed on 24.04?

**Reply by afbjorklund** [reply]
There are some links above, you can install it in a virtual machine (like on macOS or Windows) or you can build it from source.

**Reply by GongT** [reply]
This guy has compiled one: https://github.com/mgoltzsche/podman-static

**Comment by freesocket** [comment]
Or install Incus - https://linuxcontainers.org/incus/docs/main/installing/.
Then install fedora/40.
Fedora 40 has podman-5.2.2-1.fc40 up to date package.
In the Fedora container install podman.

One bonus if you use Ubuntu and Incus with zfs is automatic compression, saving disk space.
(For e.g. one of my test containers (Debian), "zfs get compressratio Data/rpool/incus/containers/pod-1, Data/rpool/incus/containers/pod-1  compressratio  2.24x").

With systemcontainers You also get more portability. I usually just blow my Ubuntu installation, install a  fresh Ubuntu, then install necessary packages (e.g. Incus, etc), then import the zpool with data and containers. The os you can always replace, but data is more important.

**Comment by muety** [comment]
Having the same problem here. Why is there no convenient way to install latest stable Podman on Ubuntu (22.04) from some trusted repo? I'd appreciate if the Podman team could provide an official DEB package.

**Reply by lstolcman** [reply]
> Having the same problem here. Why is there no convenient way to install latest stable Podman on Ubuntu (22.04) from some trusted repo? I'd appreciate if the Podman team could provide an official DEB package.

not sure whether it is trusted, but there is an alternative repo where latest podman can be installed in latest ubuntu version:
https://github.com/containers/podman/discussions/17362#discussioncomment-9635248

**Comment by junminhong** [comment]
I recently compiled a guide on installing the latest version of Podman on Ubuntu 24.04, ensuring smooth operation. If you're interested, feel free to check it out:
[How to Install the Latest Podman on Ubuntu 24.04](https://library.wowkit.net/install-latest-podman-ubuntu-24-04#installation)

**Comment by flash-me** [comment]
Just use the alvistack project from the opensuse build service
https://software.opensuse.org/download/package?package=podman&project=home%3Aalvistack 

cheers
flash ⚡

**Reply by muety** [reply]
Not recommendable. Using alvistack messed up all of my dependencies as it not only provides Podman, but hundreds of other packages too, which override those from the official repo. It was a nightmare to revert back to default.

**Reply by flash-me** [reply]
![image](https://github.com/user-attachments/assets/0d2f98a9-2327-4a3a-a8f0-713958217caa)

You can still just use these

**Reply by sbrivio-rh** [reply]
Or [these](https://tracker.debian.org/pkg/podman).

**Reply by teward** [reply]
> Or [these](https://tracker.debian.org/pkg/podman).

Yeah, those don't work in Ubuntu 24.04.  Mismatched `go` versions.  Also, I believe Ubuntu 24.04's golang is *too* old to run Podman latest.  If you look at https://github.com/containers/podman/blob/main/go.mod#L6, it requires 1.24.6 or higher. Which is *not* available in Ubuntu 24.04.  And it's not easy to backport golang inside 24.04 unless you *REALLY* know what you're doing.  It's easier to install newer golang locally with the Golang instructions (I have 1.25.3 on my computer currently, but 1.25.4 is latest and SHOULD work), and then build-from-source locally with the instructions on the Podman site. Then, I go and run the final `sudo make install` commands afterwards with `checkinstall` so that it makes a .deb that I can install with.  It's an extra step but it'll make it easy for us to delete from my system.

Ultimately, also, you *should not* use Debian packages in Ubuntu because of mismatched components, etc. as well - and I speak from experience on that front.  I would *not* use those from Debian right now.

**Comment by kamikaze** [comment]
even on Ubuntu 25.10 podman is waaaay outdated. Ancient podman compose just has so many issues...
